### ISSUE TITLE
MqttRxClient appears to be both connected and disconnected after publish is rejected due to permissions

### ISSUE THREAD

**Issue by tchupp** [hivemq/hivemq-mqtt-client#303]
## Background

We have a set of tests that guarantee our topic level permissions are working properly on our HiveMQ cluster.

Our client is configured to use the default auto reconnect config.

When we publish a message to a topic we do not have permissions for, we get:
```
com.hivemq.client.mqtt.exceptions.MqttSessionExpiredException: Session expired as connection was closed.
    Caused by: com.hivemq.client.mqtt.exceptions.ConnectionClosedException: Server closed connection without DISCONNECT.
```
Which is expected!

## Actual behavior
At this point we can still see our ClientId connected in the HiveMQ admin dashboard.

In our test, we then call MqttRxClient.disconnect() and it produces:
```
com.hivemq.client.mqtt.exceptions.MqttClientStateException: MQTT client is not connected.
```
At this point we can **still** see our ClientId connected in the HiveMQ admin dashboard!!

Following calls to MqttRxClient.connect() produce:
```
com.hivemq.client.mqtt.exceptions.MqttClientStateException: MQTT client is already connected or connecting.
```

Subsequent calls to MqttRxClient.disconnect() produce:
```
com.hivemq.client.mqtt.exceptions.MqttClientStateException: MQTT client is not connected.
```
## Expected behavior

When the client attempts to publish a message to to a topic it does not have permissions for, and is subsequently reconnected, calling `MqttRxClient.disconnect()` should properly disconnect.

This behavior is only present when using the auto reconnect.
We didn't attempt to reproduce this when forcing the client to disconnect any other way.

### Steps

1. HiveMQ has some permissions that prevent publishes
2. Configure HiveMQ client with auto-reconnect
3. Connect to HiveMQ
4. Publish to forbidden topic
5. call `disconnect` on client

### Reproducer code

```java
        Completable connectFlow = client.connect()
            .ignoreElement()

        Mqtt3Publish publishMessage = Mqtt3Publish.builder()
            .topic(MQTT_UNAUTHORIZED_PUBLISH_TOPIC)
            .payload(topicAuthorizationFailScenarioId.toString().toByteArray())
            .qos(MqttQos.EXACTLY_ONCE)
            .build();

        Completable publishFlow = client.publish(Flowable.just(publishMessage))
            .doOnNext ( result ->
                result.error
                    .ifPresentOrElse(
                        metrics::recordPublishTopicAuthorizationSuccessful,
                        metrics::recordPublishTopicAuthorizationFailed
                    )
            )
            .doOnError(it -> Logger.info("Error while publishing", it))
            .firstOrError()
            .ignoreElement();

        Completable disconnectFlow =
            client.disconnect()
                .doOnComplete (() -> Logger.info("MQTT disconnect succeeded") )
                .doOnError ( it -> Logger.info("MQTT disconnect failed", it) )
                .onErrorComplete();

        Completable redisconnectFlow =
            connectFlow
                .doOnComplete (() -> Logger.info("MQTT re-connect succeeded") )
                .doOnError (it -> Logger.info("MQTT re-connect failed", it) )
                .andThen(disconnectFlow)
                .onErrorResumeNext (__ ->
                    client.disconnect()
                        .doOnComplete (() -> Logger.info("MQTT re-disconnect succeeded") )
                        .doOnError (it -> Logger.info("MQTT re-disconnect failed", it) )
                )
                .onErrorComplete();

        return connectFlow
            .andThen(publishFlow)
            .andThen(redisconnectFlow)
            .onErrorResumeNext (() -> redisconnectFlow );
```

## Details
- Affected HiveMQ MQTT Client version(s):  1.1.0
- Used JVM version: Java 9, 10, 11

**Comment by SgtSilvio** [comment]
The client is actually not both connected and disconnected.
Its state is DISCONNECTED_RECONNECT, but connect treats this currently as connected.
Created an improvement issue for that #306

**Comment by SgtSilvio** [comment]
I will close this issue, will be resolved by #306 in the 1.2 release

**Comment by CJure** [comment]
I still have similar problem in 1.3. 


```
05:54:34.047 [RxComputationThreadPool-1] INFO  mqtt.MqttPahoClient - openConnection() start
05:54:34.123 [RxComputationThreadPool-4] INFO  mqtt.MqttPahoClient - openConnection() end
05:54:34.124 [RxComputationThreadPool-4] INFO  mqtt.MqttPahoClient - connAck=null
05:54:34.124 [RxComputationThreadPool-4] ERROR mqtt.MqttPahoClient - connect error
05:54:34.124 [RxComputationThreadPool-4] ERROR mqtt.MqttPahoClient - couldn't connect to broker={}
com.hivemq.client.mqtt.exceptions.MqttClientStateException: MQTT client is already connected or connecting.
```

If I restart program everything is fine.
